### PR TITLE
ENH: Add new GitHub community issue templates.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add new GitHub community issue templates.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Following [Matt's suggestion](https://github.com/InsightSoftwareConsortium/ITK/pull/522#issuecomment-465257439).

A few comments:
- I took the liberty of **adding the following labels**:
  - `type:Compiler`
  - `type:Coverage`
  - `type:Design`
  - `type:Testing`

- I may not have found the appropriate the **words to describe** a deficiency, weakness or improvement area, so I used *impact*, but I'm not satisfied with it. If anyone has a better suggestion, please go ahead.
- I tried to be **clear in the template descriptions**, and I think they **serve for a purpose**. That being said, if we could **use icons** or emojis like they do in the [PowerShell repository templates](https://github.com/PowerShell/PowerShell/issues/new/choose), the list would be a little bit **nicer**. However, I did many, many attempts in a private repository to add them but the templates are either not being recognized again by GitHub or the icons are simply not rendered.

File: .github/ISSUE_TEMPLATE/compiler_impact_report.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+name: Compiler Impact Report
```
Comment (thewtex) [inline .github/ISSUE_TEMPLATE/compiler_impact_report.md]: "Compiler Impact" -> "Compiler Support" may be more specific (along with file name)

File: .github/ISSUE_TEMPLATE/coverage_impact_report.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+---
+name: Coverage Impact Report
```
Comment (thewtex) [inline .github/ISSUE_TEMPLATE/coverage_impact_report.md]: "Coverage Impact Report" -> "Improve Coverage Report"

File: .github/ISSUE_TEMPLATE/coverage_impact_report.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+<!-- Before submitting an issue, please check that your issue has not been
+already filed. -->
+
+### Description
+
+<!-- Description of the impact of the current coverage in a given code area, and
+how an increased coverage would benefit the toolkit. -->
+
+### Steps to Reproduce
```
Comment (thewtex) [inline .github/ISSUE_TEMPLATE/coverage_impact_report.md]: I don't think we need many of these section for the coverage report -- they will largely be unused.

File: .github/ISSUE_TEMPLATE/regression_bug_report.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+---
+name: Regression Impact Report
+labels: type:Testing
```
Comment (hjmjohnson) [inline .github/ISSUE_TEMPLATE/regression_bug_report.md]: Somewhat tangential to this, but we need to have a *concise* description of the intended use of labels.  (summarize https://discourse.itk.org/t/github-issue-labels/1256/11)

< ! - - Labels have 3 categories that can be assigned "area: (blue hued label), type (yellow hued label):, status: (purple hued label)",  Please consider picking one label from each of these categories in order to add easily searchable informtion to the issue. -->

My critique of how the labeling system could be improved:

critique:  I think the "type:Bug" issue is too ambiguous in that is really overlaps with all other types.  in some respects all the other type: issues are all bugs.   

critique:  A new label category would complete the statndard sorting of importantce for "ugency:[0Critial | 1High | 2Medium | 3Low | 4Unnecessary]".  Color from bright red to white in decending color scheme.

critique:  All labels need a description.

critique: status labels should have a progression "status: [0Active | 0Blocked | 2Confirmed | 3NeedsInfo | 4Backlog]" in gradient of dark purple to light purple.

Comment (thewtex) [reply .github/ISSUE_TEMPLATE/regression_bug_report.md]: > critique: A new label category would complete the statndard sorting of importantce for "ugency:[0Critial | 1High | 2Medium | 3Low | 4Unnecessary]". Color from bright red to white in decending color scheme.

I disagree -- importance is subjective. To the reporter, it is almost often **Critical**, in their eyes. Other community members may have different priorities for an issue.  Also, we already have enough label categories ...

File: .github/ISSUE_TEMPLATE/compiler_impact_report.md (hunk 1 above)
Comment (jhlegarreta) [reply .github/ISSUE_TEMPLATE/compiler_impact_report.md]: Done in 72eae4b.

File: .github/ISSUE_TEMPLATE/coverage_impact_report.md (hunk 2 above)
Comment (jhlegarreta) [reply .github/ISSUE_TEMPLATE/coverage_impact_report.md]: Done in 72eae4b.

File: .github/ISSUE_TEMPLATE/coverage_impact_report.md (hunk 3 above)
Comment (jhlegarreta) [reply .github/ISSUE_TEMPLATE/coverage_impact_report.md]: Done in 72eae4b.

**Comment by jhlegarreta** [comment]
@hjmjohnson 's suggestions', along with the changes proposed in [InsightSoftwareConsortium/pulls](https://github.com/InsightSoftwareConsortium/.github/pull/1), have further impacts on the existence of some of these files (e.g. `type:Bug`), so I'll postpone any further changes until we reach consensus about these aspects.

IMHO, although virtually all can be considered as bugs, having a default `type:Bug` and associated template is not harmful. But not a strong preference.

As for the priorities, they were discarded when labels were first created a few months ago. I do not have a preference :question: If labels could be reserved for the admins, they would allow the admins to do the triage.

File: .github/ISSUE_TEMPLATE/regression_bug_report.md (hunk 4 above)
Comment (jhlegarreta) [reply .github/ISSUE_TEMPLATE/regression_bug_report.md]: All current labels **have a description now**.

As for the **importance or urgency and status** labels, I do not have a strong opinion, even if it was me that brought the issue to the table 😊 . I think that they can be useful at times, but I'm not sure whether people are likely to embrace them due to the overhead they bring (even to the maintainers -e.g. an issue is marked as critical, but it turns out it is not), or their power may be somehow lost if used only sparingly.

We may add a notice in the templates to tell users to be gentle when marking the importance/urgency if this bring us to a meeting point.

As for the colors, although I think the current code is visual enough, the 3-color code suggestions seems fair.

Again, mine is not a strong opinion, so I will embrace whatever you the heavy ITK'ers decide.

Comment (hjmjohnson) [reply .github/ISSUE_TEMPLATE/regression_bug_report.md]: The urgency was a suggestion based on using labels for project management.  Don't worry about them now, it is not a priority for ITK at the moment.

**Comment by hjmjohnson** [comment]
@jhlegarreta Please do not put this on hold.  Please take it to completion if possible.  I'll retract all my comments if they are going to hold these improvements up.  "Don't let perfection be the enemy of the good."  What you have is good, it should benefit the community as is.

**Pull request merged**